.github/workflows/docs.yaml

```diff
@@ -5,30 +5,49 @@ on:
       - main
     tags:
       - 'v*'
+  pull_request:
 
 jobs:
   deploy:
+    name: Hugo
     runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}/go
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/rotationalio/ensign/docs
     steps:
       - name: Checkout Code
         uses: actions/checkout@v3
         with:
+          path: ${{ env.GOPATH }}/src/github.com/rotationalio/ensign
           submodules: true   # Fetch Hugo themes
           fetch-depth: 0     # Fetch all history for .GitInfo and .Lastmod
 
       - name: Setup Hugo
         uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.101.0'
+          hugo-version: '0.111.0'
           extended: true
 
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18.x.x"
+
+      - name: Install NPM Dependencies
+        run: npm install
+
+      - name: Update Hugo Mod
+        run: hugo mod clean && hugo mod get -u ./... && hugo mod tidy
+
       - name: Build
-        run: hugo --minify -s docs/
+        run: hugo --gc --minify --templateMetrics --templateMetricsHints --verbose --verboseLog --buildDrafts --buildExpired --buildFuture --forceSyncStatic || hugo --themesDir ../.. --gc --minify --templateMetrics --templateMetricsHints --verbose --verboseLog --buildDrafts --buildExpired --buildFuture --forceSyncStatic
```

Comment:
These commands were from the theme `package.json` -- not sure what they do, but adding them to see if it helps 🤷🏽